WikiCohereEnglishEmbeddingOnlyScenario.cs

```diff
@@ -15,128 +27,23 @@ public override void Setup()
-
-            if(runIngestion) 
-            {
-                int totalVectors = Convert.ToInt32(this.Configurations["AppSettings:scenario:sliceCount"]);
-                await PerformIngestion(IngestionOperationType.Insert, 0 /* startVectorId */, totalVectors);
-            }
-
-            bool runQuery = Convert.ToBoolean(this.Configurations["AppSettings:scenario:runQuery"]);
-
-            if(runQuery)
-            {
-                bool performWarmup = Convert.ToBoolean(this.Configurations["AppSettings:scenario:warmup:enabled"]);
-                if (performWarmup)
-                {
-                    int numWarmupQueries = Convert.ToInt32(this.Configurations["AppSettings:scenario:warmup:numWarmupQueries"]);
-                    Console.WriteLine($"Performing {numWarmupQueries} queries for Warmup.");
-                    await PerformQuery(true /* isWarmup */, numWarmupQueries, 10 /*KVal*/, GetBaseDataPath());
-                }
-
-                int totalQueryVectors = BigANNBinaryFormat.GetBinaryDataHeader(GetQueryDataPath()).Item1;
-                for (int kI = 0; kI < K_VALS.Length; kI++)
-                {
-                    Console.WriteLine($"Performing {totalQueryVectors} queries for Recall/RU/Latency stats for K: {K_VALS[kI]}.");
-                    await PerformQuery(false /* isWarmup */, totalQueryVectors, K_VALS[kI] /*KVal*/, GetQueryDataPath());
-                }
-            }
-        }
-
-        public override void Stop()
-        {
-            bool runQuery = Convert.ToBoolean(this.Configurations["AppSettings:scenario:runQuery"]);
-            bool computeRecall = Convert.ToBoolean(this.Configurations["AppSettings:scenario:computeRecall"]);
-
-            if (runQuery && computeRecall)
-            {
-                Console.WriteLine("Computing Recall.");
-                GroundTruthValidator groundTruthValidator = new GroundTruthValidator(
-                    GroundTruthFileType.Binary,
-                    GetGroundTruthDataPath());
-
-                for (int kI = 0; kI < K_VALS.Length; kI++)
-                {
-                    int kVal = K_VALS[kI];
-                    float recall = groundTruthValidator.ComputeRecall(kVal, this.queryRecallResults[kVal]);
-
-                    Console.WriteLine($"Recall for K = {kVal} is {recall}.");
-                }
-            }
-
-            bool computeLatencyAndRUStats = Convert.ToBoolean(this.Configurations["AppSettings:scenario:computeLatencyAndRUStats"]);
-            if (computeLatencyAndRUStats)
-            {
-                bool runIngestion = Convert.ToBoolean(this.Configurations["AppSettings:scenario:runIngestion"]);
-                ComputeLatencyAndRUStats(runIngestion, runQuery);
-            }
-        }
-
-        private void ComputeLatencyAndRUStats(bool runIngestion, bool runQuery)
+        private static (int, int) ComputeInitialAndFinalThroughput(IConfiguration configurations)

```

Comment:
For completeness, allow override here as well if config 'cosmosContainerRUInitial' / 'cosmosContainerRUFinal' are populated.